src/tir/transforms/hoist_if_then_else.cc

```diff
@@ -0,0 +1,374 @@
+      is_recorder_on = false;
+    }
+    ordered_for_list_.clear();
+    var_for_map_.clear();
+    hoist_for_if_recorder = std::make_tuple(false, nullptr, nullptr);
+  }
+
+  bool RecordingComplete() {
+    if (std::get<0>(hoist_for_if_recorder)) return true;
+    return false;
+  }
+
+  const ForNode* GetTargetForNode() { return std::get<1>(hoist_for_if_recorder); }
+
+  const IfThenElseNode* GetTargetIfNode() { return std::get<2>(hoist_for_if_recorder); }
+
+ private:
+  bool CheckValidIf() {
+    // If no if var list is present, then all the condition vars are possibly from AttrStmt, so stop
+    // hoisting
+    if (if_var_list_.size() == 0) {
+      return false;
+    }
+    if (CheckAttrVar()) {
+      return false;
+    }
+    return true;
+  }
+
+  bool IsParentForLoop(int loop_pos) {
+    // Check if the loop position is higher than the parent loop position
+    for (auto var : if_var_list_) {
+      if (GetParentLoopPos(var_for_map_[var]) >= loop_pos) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  int GetParentLoopPos(const Object* node) {
+    for (size_t i = 0; i < ordered_for_list_.size(); ++i) {
+      if (ordered_for_list_[i] == node) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  void InitRecorder() { hoist_for_if_recorder = std::make_tuple(false, nullptr, nullptr); }
+
+  void StopRecording() { is_recorder_on = false; }
+
+  bool IsRecordingOn() { return is_recorder_on; }
+
+  void StartOrAddRecord(const ForNode* op) {
+    is_recorder_on = true;
+    if (!var_for_map_.count(op->loop_var.get())) {
+      var_for_map_.insert({op->loop_var.get(), op});
+    }
+    ordered_for_list_.emplace_back(op);
+  }
+
+  void RemoveRecord(const ForNode* op) {
+    StopRecording();
+    var_for_map_.erase(op->loop_var.get());
+    if (ordered_for_list_.size() > 0) ordered_for_list_.pop_back();
+  }
+
+  void StopAndAddRecord(const ForNode* for_node, const IfThenElseNode* if_node) {
+    hoist_for_if_recorder = std::make_tuple(true, for_node, if_node);
+    StopRecording();
+  }
+
+  void UpdateAttrVarList(const AttrStmtNode* op) {
+    if (const auto* iv = op->node.as<IterVarNode>()) {
+      attr_var_list_.insert(iv->var.get());
+    } else if (const auto* iv = op->node.as<VarNode>()) {
+      attr_var_list_.insert(iv);
+    }
+  }
+
+  void RemoveAttrVarList(const AttrStmtNode* op) {
+    if (const auto* iv = op->node.as<IterVarNode>()) {
+      attr_var_list_.erase(iv->var.get());
+    } else if (const auto* iv = op->node.as<VarNode>()) {
+      attr_var_list_.erase(iv);
+    }
+  }
+
+  bool CheckAttrVar() {
+    for (auto var : if_var_list_) {
+      if (attr_var_list_.count(var)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  std::vector<const ForNode*> ordered_for_list_;
+
+  std::vector<const VarNode*> if_var_list_;
+
+  std::unordered_set<const VarNode*> attr_var_list_;
+
+  VarForMap var_for_map_;
+
+  bool is_if_cond{false};
+  bool is_recorder_on{false};
```

Comment:
```suggestion
  bool is_if_cond_{false};
  bool is_recorder_on_{false};
```

Reply:
Thanks @Hzfengsy for review!
I totally agree with your comment. Will definitely handle it.
It is just that, through out TVM base code, these kind of naming convention are followed differently.
Sometimes i get confused which one to follow :)